tests/wpunit/AssertValidSchemaTest.php

```diff
@@ -210,4 +210,55 @@ public function testContentTypesOfTaxonomy() {
 
 	}
 
+	public function testSchemaSupportsLazyLoadingTypes() {
+		add_filter( 'graphql_wp_object_type_config', function ( $config ) {
+			if ( 'TestLazyType' === $config['name'] ) {
+				// Should not be called since this type does not need to be loaded.
+				$this->assertTrue( false );
+			}
+
+			return $config;
+		}, 10, 1 );
+
+		add_filter( 'graphql_object_fields', function ( $fields, $type_name ) {
+			if ( 'TestLazyType' === $type_name ) {
+				// Should not be called since this type does not need to be loaded.
+				$this->assertTrue( false );
+			}
+
+			return $fields;
+		}, 10, 2 );
+
+		register_graphql_type(
+			'TestLazyType',
+			[
+				'fields' => function () {
+					// Should not be called since this type does not need to be loaded.
+					$this->assertTrue( false );
+
+					return [];
+				},
+			]
+		);
+
+		register_graphql_field(
+			'RootQuery',
+			'example',
+			[ 'type' => 'TestLazyType' ]
```

Comment:
I think we'll need to also add a test for cases where fields are of wrapping Types. 

like `'type' => [ 'list_of' => [ 'non_null' => 'TestLazyType' ] ]`

When I was exploring fixing this issue I believe I was running into something similar to: https://github.com/webonyx/graphql-php/issues/779

It could have been caused by something else I hadn't yet addressed in the WPGraphQL codebase, though. 

I'll explore this today.

Reply:
Added!

Reply:
Amazing! Thanks so much! 🙌🏻